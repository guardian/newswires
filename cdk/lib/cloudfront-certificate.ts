import { GuStack, type GuStackProps } from '@guardian/cdk/lib/constructs/core';
import type { App } from 'aws-cdk-lib';
import { aws_certificatemanager as acm } from 'aws-cdk-lib';
import { appName } from './constants';

export type CloudFrontProps = GuStackProps & {
	domainName: string;
};

export class NewswiresCloudfrontCertificate extends GuStack {
	constructor(scope: App, id: string, props: CloudFrontProps) {
		super(scope, id, {
			app: appName,
			env: {
				region: 'us-east-1',
			},
			crossRegionReferences: true,
			...props,
		});

		new acm.Certificate(this, `cloudfront-certificate-${this.stage}`, {
			domainName: props.domainName,
			validation: acm.CertificateValidation.fromDns(),
		});
	}
}
