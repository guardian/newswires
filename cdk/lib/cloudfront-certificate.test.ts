import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { NewswiresCloudfrontCertificate } from './cloudfront-certificate';

describe('NewswiresCloudFrontCertificate', () => {
	it('matches the snapshot', () => {
		const app = new App();

		const stack = new NewswiresCloudfrontCertificate(
			app,
			'NewswiresCloudFrontCertificateTest',
			{
				stack: 'editorial-feeds',
				stage: 'TEST',
				domainName: 'newswires.TEST.dev-gutools.co.uk',
				env: { region: 'us-east-1' },
			},
		);

		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
});
