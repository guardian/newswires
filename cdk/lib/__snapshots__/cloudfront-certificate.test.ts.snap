// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NewswiresCloudFrontCertificate matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [],
    "gu:cdk:version": "TEST",
  },
  "Resources": {
    "CloudFrontCertificateArn4A1CDCF9": {
      "Properties": {
        "Description": "The ARN of the CloudFront certificate for newswires.TEST.dev-gutools.co.uk, for consumption by the Newswires app stack.",
        "Name": "/TEST/editorial-feeds/newswires/cloudfront/certificateArn",
        "Tags": {
          "App": "newswires",
          "Stack": "editorial-feeds",
          "Stage": "TEST",
          "gu:cdk:version": "TEST",
          "gu:repo": "guardian/newswires",
        },
        "Type": "String",
        "Value": {
          "Ref": "cloudfrontcertificateTEST727745A8",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "cloudfrontcertificateTEST727745A8": {
      "Properties": {
        "DomainName": "newswires.TEST.dev-gutools.co.uk",
        "Tags": [
          {
            "Key": "App",
            "Value": "newswires",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/newswires",
          },
          {
            "Key": "Name",
            "Value": "NewswiresCloudFrontCertificateTest/cloudfront-certificate-TEST",
          },
          {
            "Key": "Stack",
            "Value": "editorial-feeds",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
    },
  },
}
`;
